src/Float.mo

```diff
@@ -507,25 +499,25 @@ module {
   ///
   /// Special cases:
   /// ```
-  /// equalWithin(+0.0, -0.0, epsilon) => true for any `epsilon >= 0.0`
-  /// equalWithin(-0.0, +0.0, epsilon) => true for any `epsilon >= 0.0`
-  /// equalWithin(+inf, +inf, epsilon) => true for any `epsilon >= 0.0`
-  /// equalWithin(-inf, -inf, epsilon) => true for any `epsilon >= 0.0`
-  /// equalWithin(x, NaN, epsilon)     => false for any x and `epsilon >= 0.0`
-  /// equalWithin(NaN, y, epsilon)     => false for any y and `epsilon >= 0.0`
+  /// equal(+0.0, -0.0, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(-0.0, +0.0, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(+inf, +inf, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(-inf, -inf, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(x, NaN, epsilon)     => false for any x and `epsilon >= 0.0`
+  /// equal(NaN, y, epsilon)     => false for any y and `epsilon >= 0.0`
   /// ```
   ///
   /// Example:
   /// ```motoko
   /// import Float "mo:base/Float";
   ///
   /// let epsilon = 1e-6;
-  /// Float.equalWithin(-12.3, -1.23e1, epsilon) // => true
+  /// Float.equal(-12.3, -1.23e1, epsilon) // => true
   /// ```
-  public func equalWithin(x : Float, y : Float, epsilon : Float) : Bool {
+  public func equal(x : Float, y : Float, epsilon : Float) : Bool {
```

Comment:
@luc-blaeser are you ok with renaming equalWIthin (and notEqualWithin) to equal and notEqual?
Also another option might be to make both of these curried functions, that take the epsilon as a first argument and return an equality/inequality function of the usual binary type:

```
  public func equalWithin(epsilon: Float) : (x : Float, y : Float) - > Bool =
       func(x : Float, y: float)  { ... epsilon ...  }
  }
 (and similarly for nonEqualWithin 
```

Reply:
The motivation here is for API uniformity. I suppose we could curry the arguments, although I think `equal(x : Float, y : Float, epsilon : Float)` is more consistent with how we handle the `compare` and `equal` functions for data structures.

Reply:
Right, I thinking about the situation where you want to pass the equality function as a first-class argument (which is why we define it for convenience...)

I don't have a strong opinion...